File: ProjectTest/OneLang/Python/OneLang/One/Serialization/ObjectSerializer.py
from OneLangStdLib import *
import OneLang.One.Ast.AstTypes as astTypes
import OneLang.One.Ast.Interfaces as ints
import OneLang.One.Ast.Types as types
import OneLang.index as index

class JsonSerializer:
    def __init__(self):
        pass
    
    @classmethod
    def serialize(cls, obj):
        if isinstance(obj.type, astTypes.ClassType):
            members = []
            for field in obj.type.decl.fields:
                members.append(f'''"{field.name}": {JsonSerializer.serialize(obj.get_field(field.name))}''')
            return "{}" if len(members) == 0 else f'''{{\n{pad(/* TODO: UnresolvedMethodCallExpression */ members.join(",\n"))}\n}}'''
        else:
            return "\"<UNKNOWN-TYPE>\""

File: ProjectTest/OneLang/PHP/One/Serialization/ObjectSerializer.php
<?php

namespace One\Serialization\ObjectSerializer;

use One\Ast\AstTypes\ClassType;
use One\Ast\Interfaces\IType;
use One\Ast\Types\Class_;
use One\Ast\Types\Package;

class JsonSerializer {
    static function serialize($obj) {
        if ($obj->type instanceof ClassType) {
            $members = array();
            foreach ($obj->type->decl->fields as $field)
                $members[] = "\"" . $field->name . "\": " . JsonSerializer::serialize($obj->getField($field->name));
            return count($members) === 0 ? "{}" : "{\n" . Global.pad(/* TODO: UnresolvedMethodCallExpression */ $members->join(",\n")) . "\n}";
        }
        else
            return "\"<UNKNOWN-TYPE>\"";
    }
}
